### PR TITLE
enable vmx for vm factory

### PR DESCRIPTION
**Pull request by bergwolf**

### CODE REVIEW

**Review by liubin — APPROVED** [review]

File: src/runtime/virtcontainers/qemu_amd64.go [hunk 1]
```diff
@@ -38,6 +38,8 @@ const (
 
 	defaultQemuMachineOptions = "accel=kvm,kernel_irqchip=on"
 
+	splitIrqChipMachineOptions = "accel=kvm,kernel_irqchip=split"
+
```
Comment (gkurz) [inline src/runtime/virtcontainers/qemu_amd64.go:42]: This change doesn't seem to belong here. Please move it to another patch.

**Review by gkurz — COMMENTED** [review]
FYI, migration of L1 with running L2s is supposed to _work_ with recent kernels according to https://bugzilla.kernel.org/show_bug.cgi?id=53851#c3 .

File: src/runtime/virtcontainers/qemu_ppc64le.go [hunk 2]
```diff
@@ -115,10 +115,6 @@ func (q *qemuPPC64le) bridges(number uint32) {
 	q.Bridges = genericBridges(number, q.qemuMachine.Type)
 }
 
-func (q *qemuPPC64le) cpuModel() string {
-	return defaultCPUModel
-}
-
```
Comment (gkurz) [inline src/runtime/virtcontainers/qemu_ppc64le.go:121]: This change makes sense but it isn't really related to this PR either.

**Comment by arronwy** [comment]
Can one of the admins verify this patch?

**Comment by jodh-intel** [comment]
Ping @bergwolf.

File: src/runtime/virtcontainers/qemu_amd64.go (hunk 1 above)
Comment (bergwolf) [reply src/runtime/virtcontainers/qemu_amd64.go:42]: It's a simple cleanup. I've split it into a separate commit.

File: src/runtime/virtcontainers/qemu_ppc64le.go (hunk 2 above)
Comment (bergwolf) [reply src/runtime/virtcontainers/qemu_ppc64le.go:120]: Same here. It's simple enough to go through in the same PR but in a separate commit.

**Comment by bergwolf** [comment]
rebased on the latest main.

**Comment by bergwolf** [comment]
@gkurz The intention is not to migrate from L1 to L2, but from L1 to L1 with vmx capability enabled.

**Review by teawater — APPROVED** [review]
